From: Hannes Reinecke <hare@suse.de>
Date: Wed, 29 Oct 2014 14:51:49 +0100
Subject: [PATCH] megaraid_sas: Disable fastpath writes for non-RAID0
References: bnc#897502
Patch-Mainline: not yet

Fastpath writes only work for RAID0, so disable it for
other configurations.


Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/megaraid/megaraid_sas_fp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/scsi/megaraid/megaraid_sas_fp.c b/drivers/scsi/megaraid/megaraid_sas_fp.c
index 7788ee6..01a668e 100644
--- a/drivers/scsi/megaraid/megaraid_sas_fp.c
+++ b/drivers/scsi/megaraid/megaraid_sas_fp.c
@@ -370,7 +370,8 @@ MR_BuildRaidContext(struct megasas_instance *instance,
 					       fpReadAcrossStripe));
 		else
 			io_info->fpOkForIo = (raid->capability.fpWriteCapable &&
-					      ((num_strips == 1) ||
+					      ((num_strips == 1 &&
+						raid->level == 0) ||
 					       raid->capability.
 					       fpWriteAcrossStripe));
 	} else
-- 
1.8.5.2

